swap/src/cli/list_sellers.rs

```diff
@@ -164,30 +165,42 @@ impl EventLoop {
                                 self.reachable_asb_address.insert(peer_id, address.clone());
                             }
                         }
-                        SwarmEvent::UnreachableAddr { peer_id, error, address, .. } => {
-                            if address == self.rendezvous_addr {
+                        SwarmEvent::OutgoingConnectionError { peer_id, error } => {
+                            if let Some(peer_id_from_error) = peer_id {
+
+                                if &peer_id_from_error == &self.rendezvous_peer_id {
                                 tracing::error!(
                                     "Failed to connect to rendezvous point at {}: {}",
-                                    address,
+                                    &self.rendezvous_addr,
                                     error
                                 );
 
                                 // if the rendezvous node is unreachable we just stop
                                 return Vec::new();
+                            } else {
+                                tracing::error!(
+                                    "You connected to the wrong Peer: {} Error: {}",
+                                    &peer_id_from_error,
+                                    error
+                                );
+                                     // if for some reason the peer is not the same it will return empty too.
+                                     // this is just for the CLI to not get stuck
+                                     return Vec::new();
+                            }
                             } else {
                                 tracing::debug!(
                                     "Failed to connect to peer at {}: {}",
-                                    address,
+                                    &self.rendezvous_addr,
                                     error
                                 );
-                                self.unreachable_asb_address.insert(peer_id, address.clone());
+                                self.unreachable_asb_address.insert(self.rendezvous_peer_id, self.rendezvous_addr.clone());
```

Comment:
Same thing. I think the peer might be an asb. We are adding a rendezvous point here to a list called `self.unreachable_asb_address ` which doesn't really make sense.

Reply:
Agreed, fixed in same commit: 7838442947c471ce6deb91f19e989bd636f6f076

Reply:
Awesome, thanks!